schema/samcli.json

```diff
@@ -143,9 +143,175 @@
+                  "title": "image_repositories",
+                  "type": "string",
+                  "description": "Mapping of Function Logical ID to AWS ECR Repository URI.\n\nExample: Function_Logical_ID=ECR_Repo_Uri\nThis option can be specified multiple times."
+                },
+                "force_upload": {
+                  "title": "force_upload",
+                  "type": "boolean",
+                  "description": "Indicates whether to override existing files in the S3 bucket. Specify this flag to upload artifacts even if they match existing artifacts in the S3 bucket."
+                },
+                "s3_prefix": {
+                  "title": "s3_prefix",
+                  "type": "string",
+                  "description": "Prefix name that is added to the artifact's name when it is uploaded to the AWS S3 bucket."
+                },
+                "kms_key_id": {
+                  "title": "kms_key_id",
+                  "type": "string",
+                  "description": "The ID of an AWS KMS key that is used to encrypt artifacts that are at rest in the AWS S3 bucket."
+                },
+                "role_arn": {
+                  "title": "role_arn",
+                  "type": "string",
+                  "description": "ARN of an IAM role that AWS Cloudformation assumes when executing a deployment change set."
+                },
+                "use_json": {
+                  "title": "use_json",
+                  "type": "boolean",
+                  "description": "Indicates whether to use JSON as the format for the output AWS CloudFormation template. YAML is used by default."
+                },
+                "resolve_s3": {
+                  "title": "resolve_s3",
+                  "type": "boolean",
+                  "description": "Automatically resolve AWS S3 bucket for non-guided deployments. Enabling this option will also create a managed default AWS S3 bucket for you. If one does not provide a --s3-bucket value, the managed bucket will be used. Do not use --guided with this option."
+                },
+                "resolve_image_repos": {
+                  "title": "resolve_image_repos",
+                  "type": "boolean",
+                  "description": "Automatically create and delete ECR repositories for image-based functions in non-guided deployments. A companion stack containing ECR repos for each function will be deployed along with the template stack. Automatically created image repositories will be deleted if the corresponding functions are removed."
+                },
+                "metadata": {
+                  "title": "metadata",
+                  "type": "string",
+                  "description": "Map of metadata to attach to ALL the artifacts that are referenced in the template."
+                },
+                "notification_arns": {
+                  "title": "notification_arns",
+                  "type": "array",
```

Comment:
can we set items section to the schema here to let customer knows that the values of the list items should be string?

Reply:
Yup, I was unaware of that parameter for arrays, we can include that as well 👍